testsuite/integration/microprofile/src/test/java/org/wildfly/test/integration/observability/TestContainersEnricher.java

```diff
@@ -0,0 +1,69 @@
+package org.wildfly.test.integration.observability;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.jboss.arquillian.test.spi.TestEnricher;
+import org.jboss.as.test.shared.util.AssumeTestGroupUtil;
+import org.wildfly.test.integration.observability.container.OpenTelemetryCollectorContainer;
+
+public class TestContainersEnricher implements TestEnricher {
+    private static final ThreadLocal<OpenTelemetryCollectorContainer> otelCollectorContainer = new ThreadLocal<>();
+
+    @Override
+    public void enrich(Object testCase) {
+        if (AssumeTestGroupUtil.isDockerAvailable()) {
+            for (Field field : getFieldsWithAnnotation(testCase.getClass())) {
+                OpenTelemetryCollectorContainer value = lookup(field.getType());
+                try {
+                    if (!field.canAccess(testCase)) {
+                        field.setAccessible(true);
+                    }
+                    field.set(testCase, value);
+                } catch (Exception e) {
+                    throw new RuntimeException("Could not set value on field " + field + " using " + value, e);
+                }
+            }
+        }
+    }
+
+    private OpenTelemetryCollectorContainer lookup(Class<?> type) {
+        if (type.isAssignableFrom(OpenTelemetryCollectorContainer.class)) {
+
+            OpenTelemetryCollectorContainer container = otelCollectorContainer.get();
+            if (container == null) {
+                container = new OpenTelemetryCollectorContainer();
+                container.start();
+                otelCollectorContainer.set(container);
```

Comment:
What calls container.stop()?

How do we know subsequent calls to enrich will be from the same thread?

Reply:
Currently, it seems that the container is cleaned up implicitly when the test JVM shuts down (ryuk maybe?). I'm digging through the Arquillian docs (and the wildfly-arquillian code) to find a way to be more explicit. I may be overthinking all of this and just need to go back to how it was, at least for now. I'm trying to find a way to get one container instance per class to help prevent data leak between test runs. Perhaps a static in the ServerSetupTask is sufficient, but might break if tests are run in multiple threads (and that concern might also be borrowing trouble). Let me think about all of this some more... :)